posthog/tasks/test/test_usage_report.py

```diff
@@ -1562,6 +1562,207 @@ def _setup_teams(self) -> None:
+
+            for _ in range(5):
+                ExternalDataJob.objects.create(
+                    team_id=3,
+                    finished_at=now(),
+                    rows_synced=10,
+                    status=ExternalDataJob.Status.COMPLETED,
+                    pipeline=source,
+                    pipeline_version=ExternalDataJob.PipelineVersion.V1,
+                )
+
+            for _ in range(5):
+                ExternalDataJob.objects.create(
+                    team_id=4,
+                    finished_at=now(),
+                    rows_synced=10,
+                    status=ExternalDataJob.Status.COMPLETED,
+                    pipeline=source,
+                    pipeline_version=ExternalDataJob.PipelineVersion.V1,
+                )
+
+            period = get_previous_day(at=now() + relativedelta(days=1))
+            period_start, period_end = period
+            all_reports = _get_all_org_reports(period_start, period_end)
+
+            assert len(all_reports) == 3
+
+            org_1_report = _get_full_org_usage_report_as_dict(
+                _get_full_org_usage_report(all_reports[str(self.org_1.id)], get_instance_metadata(period))
+            )
+
+            org_2_report = _get_full_org_usage_report_as_dict(
+                _get_full_org_usage_report(all_reports[str(self.org_2.id)], get_instance_metadata(period))
+            )
+
+            assert org_1_report["organization_name"] == "Org 1"
+            assert org_1_report["rows_synced_in_period"] == 0
+            assert org_1_report["free_historical_rows_synced_in_period"] == 100
+
+            assert org_1_report["teams"]["3"]["rows_synced_in_period"] == 0
+            assert org_1_report["teams"]["3"]["free_historical_rows_synced_in_period"] == 50
+            assert org_1_report["teams"]["4"]["rows_synced_in_period"] == 0
+            assert org_1_report["teams"]["4"]["free_historical_rows_synced_in_period"] == 50
+
+            assert org_2_report["organization_name"] == "Org 2"
+            assert org_2_report["rows_synced_in_period"] == 0
+            assert org_2_report["free_historical_rows_synced_in_period"] == 0
+
+    @patch("posthog.tasks.usage_report.get_ph_client")
+    @patch("posthog.tasks.usage_report.send_report_to_billing_service")
+    def test_external_data_rows_synced_after_free_period_response(
+        self, billing_task_mock: MagicMock, posthog_capture_mock: MagicMock
+    ) -> None:
+        self._setup_teams()
+
+        with freeze_time("2025-10-30T00:00:00Z"):
+            source_4 = ExternalDataSource.objects.create(
+                team_id=4,
+                source_id="source_id_2",
+                connection_id="connection_id_2",
+                status=ExternalDataSource.Status.COMPLETED,
+                source_type=ExternalDataSourceType.STRIPE,
+            )
+
+        with freeze_time("2025-11-07T01:00:00Z"):
+            source_3 = ExternalDataSource.objects.create(
+                team=self.analytics_team,
```

Comment:
Here also `analytics_team` has id `2` and this source is used to create jobs for team id `3`